### ISSUE TITLE
Timeout Leak from `d` function

### ISSUE THREAD

**Issue by kerrigan-nm** [anywhichway/nano-memoize#68]
Since the `d` function [reassigns the value of the `to` timeout variable on each call](https://github.com/anywhichway/nano-memoize/blob/5a36813/src/index.js#L45), only the last timeout created is ever available in scope. This means that [when the `clear` function clears the timeout](https://github.com/anywhichway/nano-memoize/blob/5a36813/src/index.js#L101), it is only ever clearing the most recently created timeout.

Basically, if `maxAge` is set, a new timeout is created every time a memoized function is called with a new set of arguments, and all but the _last_ timeout created are unable to be cleared. This holds the node process open until those timeouts complete, even if `.clear()` is called on the memoized function.

**Comment by kerrigan-nm** [comment]
Pushed a [draft PR](https://github.com/anywhichway/nano-memoize/pull/69) with one approach to addressing the issue.

**Comment by anywhichway** [comment]
@kerrigan-nm It there a reason you opened and then closed this request? Did you determine a leak was not actually happening?

Also, it has been a long time since I was in this code. Would it be better to check and see if to is non-nul and then just clear it before setting a new timeout rather than keep a Set around.
